proof/verifier.go

```diff
@@ -356,6 +356,17 @@ func (p *Proof) verifyGroupKeyReveal() error {
 // block header is invalid (usually: not present on chain).
 type HeaderVerifier func(blockHeader wire.BlockHeader, blockHeight uint32) error
 
+// GroupVerifier is a callback function which returns an error if the given
+// group key has not been imported by the tapd daemon. This can occur if the
+// issuance proof for the group anchor has not been imported or synced.
+type GroupVerifier func(groupKey *btcec.PublicKey) error
+
+// GroupAnchorVerifier is a callback function which returns an error if the
+// given genesis is not the asset genesis of the group anchor. This would be
+// true for an reissuance into an existing group.
+type GroupAnchorVerifier func(gen *asset.Genesis,
```

Comment:
Having trouble parsing the use of this from the function signature and also comment name. IIUC, you also need the tapscript root to verify the mapping, but then this seems like just the normal genesis reveal verification?

Reply:
Similar to the group key reveal verification, but in that context we're checking across different parts of the proof, and here we're checking against the DB or re-deriving the key. Made the comment more verbose.